Add Ikemen-Go (only x86_64 at the moment)

lbrpdx opened the pull request:
Ikemen-Go is an extension to MUGEN, with more capacities (and it's an open-source re-implementation of IKEMEN in Golang). Very active in the fighting community. Installation of the games needs a little bit of work, but worth it.
It can theoretically run on a RPi4, but it needs some work to compile it at the moment. At least we have a first step here.

dmanlfc commented:
Explain - `Installation of the games needs a little bit of work, but worth it.`

lbrpdx commented:
I’ll write something in the wiki when I have time.

dmanlfc commented:
> I’ll write something in the wiki when I have time.

I need to understand how it works before accepting...

dmanlfc commented:
Also, updated files

```
config BR2_PACKAGE_IKEMEN
    bool "ikemen"
    depends on BR2_PACKAGE_HOST_GO_TARGET_ARCH_SUPPORTS
    depends on BR2_PACKAGE_XORG7
    select BR2_PACKAGE_MESA3D
    select BR2_PACKAGE_LIBGTK3
    select BR2_PACKAGE_OPENAL
    select BR2_PACKAGE_LIBGLFW
    help
        A MUGEN emulator written in GoLang

comment "ikemen needs a toolchain w/ threads"
	depends on BR2_PACKAGE_HOST_GO_TARGET_ARCH_SUPPORTS
```

```
################################################################################
#
# ikemen
#
################################################################################
# Last commit on Oct 27, 2022
IKEMEN_VERSION = f586e49413c26646642dc3b5039478bd764d386a
IKEMEN_SITE = https://github.com/ikemen-engine/Ikemen-GO
IKEMEN_LICENSE = MIT
IKEMEN_DEPENDENCIES = libgtk3 mesa3d openal libglfw

IKEMEN_SITE_METHOD = git
IKEMEN_GIT_SUBMODULES = YES

define IKEMEN_BUILD_CMDS
	$(HOST_GO_TARGET_ENV) $(MAKE) LD="$(TARGET_LD)" -C $(@D) -f Makefile Ikemen_GO_Linux
endef

define IKEMEN_INSTALL_TARGET_CMDS
	mkdir -p $(TARGET_DIR)/usr/bin
	$(INSTALL) -D $(@D)/bin/Ikemen_GO_Linux $(TARGET_DIR)/usr/bin/ikemen
	# evmapy
	mkdir -p $(TARGET_DIR)/usr/share/evmapy
	cp $(BR2_EXTERNAL_BATOCERA_PATH)/package/batocera/emulators/ikemen/ikemen.keys $(TARGET_DIR)/usr/share/evmapy
endef

$(eval $(golang-package))

```

This is cleaner & has the correct dependencies ^

Finally if this is a 'port' as per the select section of config.in then the path should be `ports` not `emulators`.

Darknior commented:
Excellent idea, this engine will replace our old MUGEN with Wine in time :)
I think it can be good to put them in the same GROUP in ES if we want to see only one MUGEN systeme, and not two.
Thanks

lbrpdx commented:
@dmanlfc Thanks for the review, I updated the build files accordingly... but for the build command itself, we need to change the host Golang flags anyways (otherwise mods fail), so in that version, we need even more flags. Not sure it's clearer ultimately, but it's now done anyways.

For ports vs emulators, Ikemen is an engine that runs various games, we ship the binary and we expect users to add game files. So, it's akin to OpenBOR, ScummVM, Solarus, EasyRPG... (all of them in `emulators`). I think gzdoom (in `ports`), should move there too, btw. At least from a Batocera structure perspective. For config selection, I moved it to `BR2_PACKAGE_BATOCERA_HOMEBREW_SYSTEM`.

How to test it? Simple way: download a game that is made for Ikemen-Go 0.98.2 like [Famicom Fighters](http://mcjimmy.net/famicomfighters.html). Unzip the game to `/userdata/roms/ikemen/FamicomFighters.ikemen` (extension `.ikemen` or `.pc`). Reload your games list and launch it. 

But Ikemen-Go is way more powerful and you can create you own fighting game relatively easily, reusing sprites and assets from Mugen or Ikemen. You can create your own ultimate roster pretty easily, with hundreds of characters. That's where a little bit of research and/or documentation is needed, but I can't wait to see what Batocera creators will make out of it.

@Darknior Ikemen is clearly the future of Mugen. Unfortunately, today, we can't just replace Mugen with it. Maybe in the future, but I tested several Mugen games and they fail under Ikemen. They need to be ported to Ikemem like I said above. Do we want to group under Mugen, or do we think in a few months/year Ikemen will be the main one? I think we can decide later.

Darknior commented:
> @Darknior Ikemen is clearly the future of Mugen. Unfortunately, today, we can't just replace Mugen with it. Maybe in the future, but I tested several Mugen games and they fail under Ikemen. They need to be ported to Ikemem like I said above. Do we want to group under Mugen, or do we think in a few months/year Ikemen will be the main one? I think we can decide later.

For sure i'm 100% agree with you and claim it from a long time, Ikemen is the future !!!
And yes games are not compatible, creators must modify them for Ikement.
And like for some years we will must have the two engine, the old and the new, i only ask for a visual GROUP in ES, to let users having MUGEN and IKEMEN or only one systeme, MUGEN (with Ikemen) inside, because MUGEN has been a better known name for 20 years.

Darknior commented:
> For ports vs emulators, Ikemen is an engine that runs various games, we ship the binary and we expect users to add game files. So, it's akin to OpenBOR, ScummVM, Solarus, EasyRPG... (all of them in `emulators`). I think gzdoom (in `ports`), should move there too, btw. At least from a Batocera structure perspective. For config selection, I moved it to `BR2_PACKAGE_BATOCERA_HOMEBREW_SYSTEM`.

For sure GZdoom must move with others. 
But there are not emulatores, there are engine. OpenBOR, Ikemen, EasyRPG, Solarus, GZDom.
ScummVM is an emulator, AGS is an engine, now AGS is an engine included in ScummVM, ScummVM can move with engines :)
Maybe it can be a good idea to have emulators, a group PORTS, and a group ENGINE ??
Like we have now Fantasy Consoles ....
